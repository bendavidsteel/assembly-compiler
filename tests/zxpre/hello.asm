name: db "Hello world!\n"    ; message

      ; Print hello world
main: mov %a,0x04            ; sys_write
      mov %b,0x01            ; stdout
      mov %c,name            ; string to print
      mov %d,0x0d            ; length of string
      syscall                ; invoke sys call

      ; Clean exit
      mov %a,0x01            ; sys_exit
      syscall                ; invoke sys call
