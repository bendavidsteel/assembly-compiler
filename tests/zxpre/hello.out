name:    db "Hello world!\n"
main:    mov %a,0x04
         mov %b,0x01
         mov %c,name
         mov %d,0x0d
         syscall
         mov %a,0x01
         syscall
